src/linking/linkModule.ml

```diff
@@ -54,41 +54,57 @@ importing module would work:
-imports to a symbol should resolve to the same constant to support equality as
-above, and additionally pointer arithmetic.
+Exposing a data pointer generates a GOT.mem import. This is to support 
+accesses to the data and pointer equality and pointer arithmetics.
+All GOT.mem imports should resolve to the absolute pointer of their exported data. 
+The corresponding export of the GOT.mem denotes the pointer offset relative
+to the memory base of the module.
 
-(Pointer arithmetic on function pointers are undefined behavior is C and is not
+(Pointer arithmetic on function pointers are undefined behavior in C and is not
 supported by clang's wasm backend)
 
 Normally this stuff is for dynamic linking, but we want to link the RTS
-statically, so we resolve these imports during linking. Currently we only
-support GOT.func imports, but implementing GOT.mem imports would be similar.
-Secondly, we only support GOT.func imports in the module that defines the
-function that we take the address of. This currently works as moc-generated code
-doesn't import function addresses from the RTS.
+statically, so we resolve these imports during linking. We only support 
+GOT.func and GOT.mem imports in the module that defines the function that we 
+take the address of. This currently works as moc-generated code doesn't import 
+function addresses or data pointers from the RTS.
 
-We resolve GOT.func imports in two steps:
+We resolve GOT imports in two steps:
 
-- After loading the RTS module we generate a list of (global index, function
-  index) pairs of GOT.func imports. In the example above, global index is N and
-  function index is the index of f0 in the defining module (the RTS).
+- After loading the RTS module we collect GOT imports and determine their
+  implementation in the defining module (the RTS):
 
-  This is implemented in `collect_got_func_imports`.
+  For each `GOT.func` import, we determine the corresponding function index, 
+  e.g. `f0` in the example above.
+  For each `GOT.mem` import, we detemine the corresponding global storing the 
+  data offset relative to the memory base of the library.
 
-- After merging the sections we add the functions to the table and replace
-  `GOT.func` imports with globals to the functions' table indices.
+  This is implemented in `collect_got_imports`.
 
+- After merging the sections, we replace the GOT imports with globals in the 
+  module section.
+
+  For each `GOT.func`, we create an element in the table with the function index
+  determined in the first step. The `GOT.func` import is replaced by a global
+  that refers to the new table element.
+
+  For each `GOT.mem`, we replace the GOT accesses in the AST by code that computes 
+  the absolute address of the data. This is the library's memory base plus the 
+  relative data offset stored in the global as determined in the first step.  
+  The `GOT.mem` import is replaced by a dummy global that only serves to maintain 
```

Comment:
I wonder if the dummy global could instead hold the absolute/resolved address, and we replace references to the import by references to the resolved global? But I don't fully understand this stuff..

Reply:
I find this actually a good idea. If okay, I'll put it on backlog as GOT.mem are seldom.